api/stripe.go

```diff
@@ -226,6 +239,67 @@ func (api *API) stripeCheckoutPOST(u *database.User, w http.ResponseWriter, req
 	api.WriteJSON(w, response)
 }
 
+// stripeCheckoutIDGET checks the status of a checkout session. If the checkout
+// is successful and results in a higher tier sub than the current one, we
+// upgrade the user to the new tier.
+func (api *API) stripeCheckoutIDGET(u *database.User, w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
+	checkoutSessionID := ps.ByName("checkout_id")
+	subStr := "subscription"
+	params := &stripe.CheckoutSessionParams{
+		Params: stripe.Params{
+			Expand: []*string{&subStr},
+		},
+	}
+	cos, err := cosession.Get(checkoutSessionID, params)
+	if err != nil {
+		api.WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	if cos.Customer.ID != u.StripeID {
+		api.WriteError(w, errors.New("checkout session does not belong to current user"), http.StatusBadRequest)
```

Comment:
Hm, I see what you mean. I would still go for Bad Request on the assumption that the caller messed up their request on accident, rather than trying to abuse the API. Kind of "assume a mistake, rather than bad intentions".

That said, I'm perfectly fine with changing this to a 403 if that seems more appropriate.

Reply:
yeah that assumption you propose is one that's usually not made in this case. 
if you try to do things using somebody else's credentials usually results in forbidden

Reply:
OK, changed.